### PR TITLE
Re: Add support for using nested DTOs

### PR DESCRIPTION
**Pull request by eltharin**
Fork for errors correction of https://github.com/doctrine/orm/pull/11238

### CODE REVIEW

**Comment by norkunas** [comment]
Your commit lost all the other authors credits :roll_eyes:

**Comment by eltharin** [comment]
Take my changes and make it to yours, you don't touch your PR since 5 month but it's a super idea.

And if you can explain me how I had to do, i'm not very comfortable with PR workflow.

**Comment by norkunas** [comment]
It's not my commits also, I've just cherry-picked to adapt for current codebase.

> you don't touch your PR since 5 month

Because I've waited to get some directions what I should do? As I see you just pushed that error to the baseline

**Comment by norkunas** [comment]
> And if you can explain me how I had to do, i'm not very comfortable with PR workflow.

`git cherry-pick commit-hash` it's a one line and then conflicts resolving if needed :)

File: psalm-baseline.xml [hunk 1]
```diff
@@ -265,9 +266,6 @@
       <code><![CDATA[setValue]]></code>
       <code><![CDATA[setValue]]></code>
     </PossiblyNullReference>
-    <PossiblyUndefinedArrayOffset>
-      <code><![CDATA[$newObject['args']]]></code>
-    </PossiblyUndefinedArrayOffset>
```
Comment (greg0ire) [inline psalm-baseline.xml:270]: If there is an issue with Psalm, you should report it to Psalm first before using `psalm.xml` to ignore the issue (and when you do, you should put the link to the bug report there). That's why I asked for a reproducer on the other ticket.

Comment (eltharin) [reply psalm-baseline.xml:270]: For This I juste replace the same probleme witch was before in ArrayHydrator and ObjectHydrator to AbstractHydrator.
I don't see ArrayHydrator and ObjectHydrator  in psalm.xml
For the reproducer I'm sorry but I don't understand want you want.

**Comment by eltharin** [comment]
@norkunas you want to make corrections to your repo or I continue here ?

**Comment by norkunas** [comment]
No prob if you'll continue

File: psalm-baseline.xml (hunk 1 above)
Comment (greg0ire) [reply psalm-baseline.xml:270]: Oh yes indeed, those are not really new violations. Should be fine then.

**Comment by greg0ire** [comment]
Please improve your commit message according to [the contributing guide](https://www.doctrine-project.org/contribute/index.html#working-on-topic-branches).

**Comment by eltharin** [comment]
and if you have 10 minutes, don't hesitate to explain to me what could have been done better by email.
For doing better last time.

**Comment by greg0ire** [comment]
I wrote the contributing guide (at least the part about the commit messages). Is there a part that is unclear? Also, I'd rather talk on doctrine.slack.com if you don't mind.

**Comment by eltharin** [comment]
sorry I read a little so fast, I think it was "just" beacause my commit was only "corrections" without any more.

**Comment by greg0ire** [comment]
I don't think people reading the history once this is merged need to know there was a need to update the baseline. Please kindly squash your commits together. If you don't, we'll try to remember to do it for you but it's best if you save us this trouble.

How to do that?

1. `git rebase -i origin/3.2.x`, assuming `origin` is a git remote that points to this repository, and not your fork. If you're not sure what your remotes are, run `git remote -vvv`, there should be your fork and the holy/reference/base/origin/whatever-you-call-it repository.
2. A window will show up with many lines, replace `pick` with `fixup` on every line but the first one
3. Close your editor, git should do its magic, and you should end up with one commit
4. Use `git push --force` to overwrite what you already push. Don't forget the `--force` option otherwise git will try to merge both things together.

**Comment by greg0ire** [comment]
The docs need to be adjusted, as they say "Note that you can only pass scalar expressions to the constructor." DTOs are not scalar expressions.

File: docs/en/reference/dql-doctrine-query-language.rst [hunk 2]
```diff
@@ -560,7 +560,34 @@ And then use the ``NEW`` DQL keyword :
...
+        }
+    }
+	
+.. code-block:: php
+
+    <?php
+    $query = $em->createQuery('SELECT NEW CustomerDTO(c.name, e.email, NEW AddressDTO(a.street, a.city, a.zip)) FROM Customer c JOIN c.email e JOIN c.address a');
+    $users = $query->getResult(); // array of CustomerDTO
+	
+Note that you can only pass scalar expressions or another Data Transfer Objects to the constructor.
```
Comment (greg0ire) [inline docs/en/reference/dql-doctrine-query-language.rst]: "Another" is singular, but objects is "plural"

**Review by greg0ire — DISMISSED** [review]

File: docs/en/reference/dql-doctrine-query-language.rst [hunk 3]
```diff
@@ -560,7 +560,34 @@ And then use the ``NEW`` DQL keyword :
...
+    {
+        public function __construct($name, $email, $address, $value = null)
+        {
+            // Bind values to the object properties.
+        }
+    }
+	
+    class AddressDTO
+    {
+        public function __construct($street, $city, $zip)
```
Comment (SenseException) [inline docs/en/reference/dql-doctrine-query-language.rst]: Please use some type hints for that example. Especially the `AddressDTO` type for `$address`.

```suggestion
        public function __construct(string $name, string $email, AddressDTO $address, string|null $value = null)
        {
            // Bind values to the object properties.
        }
    }
	
    class AddressDTO
    {
        public function __construct(string $street, string $city, string $zip)
```

**Review by SenseException — CHANGES_REQUESTED** [review]

**Review by SenseException — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @eltharin !

**Comment by greg0ire** [comment]
Argh! Wrong target branch :sweat: 
This should have targeted 3.3.x

**Comment by derrabus** [comment]
Shall I do a hard reset or do you prefer a revert? 🫣

**Comment by greg0ire** [comment]
Did the hard reset already

**Comment by greg0ire** [comment]
@eltharin please recreate this PR, but this time, target 3.3.x please. More help on this at https://github.com/doctrine/orm/issues/11208

**Comment by eltharin** [comment]
ok

**Pull request merged**